UnityProject/Assets/Scripts/Managers/SoundManager.cs

```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
 			};
 		}
 
-		sndName = Instance.ResolveSoundPattern(sndName);
-		PlaySoundMessage.Send(recipient, sndName, worldPos, polyphonic, sourceObj, shakeParameters, audioSourceParameters);
+		AddressableAudioSource addressableAudioSource = await GetAddressableAudioSourceFromCache(addressableAudioSources);
+		PlaySoundMessage.Send(recipient, addressableAudioSource, worldPos, polyphonic, sourceObj, shakeParameters, audioSourceParameters);
 	}
 
 	/// <summary>
 	/// Play a sound locally
+	/// If more than one is specified, one will be picked at random.
 	/// </summary>
-	/// <param name="name">Name of the sound to be played</param>
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, one will be picked at random.</param>
 	/// <param name="audioSourceParameters">Parameters for how to play the sound</param>
 	/// <param name="polyphonic">Should the sound be played polyphonically</param>
-	public static void Play(string name, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
+	public static async void Play(AddressableAudioSource addressableAudioSource, string soundSpawnToken, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
 	{
-		name = Instance.ResolveSoundPattern(name);
-		var sound = Instance.GetSourceFromPool(Instance.sounds[name]);
+		Play(new List<AddressableAudioSource>() { addressableAudioSource }, soundSpawnToken, audioSourceParameters, polyphonic);
+	}
 
-		ApplyAudioSourceParameters(audioSourceParameters, sound.audioSource);
+	/// <summary>
+	/// Play a sound locally
+	/// If more than one is specified, one will be picked at random.
+	/// </summary>
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, one will be picked at random.</param>
+	/// <param name="soundSpawnToken">The SoundSpawn Token that identifies the same sound spawn instance across server and clients</returns>
+	/// <param name="audioSourceParameters">Parameters for how to play the sound</param>
+	/// <param name="polyphonic">Should the sound be played polyphonically</param>
+	public static async void Play(List<AddressableAudioSource> addressableAudioSources, string soundSpawnToken, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
+	{
+		AddressableAudioSource addressableAudioSource = await GetAddressableAudioSourceFromCache(addressableAudioSources);
+		SoundSpawn soundSpawn = Instance.GetNewSoundSpawn(addressableAudioSource.AudioSource, soundSpawnToken);
+		ApplyAudioSourceParameters(audioSourceParameters, soundSpawn);
 
-		Instance.PlaySource(sound, polyphonic, true, audioSourceParameters != null && audioSourceParameters.MixerType != MixerType.Unspecified);
+		Instance.PlaySource(soundSpawn, polyphonic, true, audioSourceParameters != null && audioSourceParameters.MixerType != MixerType.Unspecified);
```

Comment:
Codacy found an issue: [Remove this default value assigned to parameter 'Global'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)